wallet/src/main/java/io/neow3j/wallet/Wallet.java

```diff
@@ -100,39 +92,91 @@ public ScryptParams getScryptParams() {
      * @throws WalletStateException if the wallet does not contain a default account.
      */
     public Account getDefaultAccount() {
-        return this.accounts.values().stream()
-                .filter(Account::isDefault)
-                .findFirst()
-                .orElseThrow(() -> new WalletStateException("The wallet does not contain a "
-                        + "default account."));
+        if (!hasDefaultAccount()) {
+            throw new IllegalStateException("Can't get default account on wallet. Wallet does "
+                    + "not contain a default Account.");
+        }
+        return this.accounts.get(this.defaultAccount);
     }
 
-    public void setName(String name) {
+    public Boolean hasDefaultAccount() {
+        return this.defaultAccount != null;
+    }
+
+    /**
+     * Checks whether an account is the default account in the wallet.
+     *
+     * @param accountScriptHash the account to be checked.
+     * @return Whether the given account is the default account in this wallet.
+     */
+    public Boolean isDefault(ScriptHash accountScriptHash) {
+        return accountScriptHash == getDefaultAccount().getScriptHash();
+    }
+
+    public Wallet setName(String name) {
         this.name = name;
+        return this;
     }
 
-    public void setVersion(String version) {
+    public Wallet setVersion(String version) {
         this.version = version;
+        return this;
+    }
+
+    public Wallet setScryptParams(ScryptParams scryptParams) {
+        this.scryptParams = scryptParams;
+        return this;
     }
 
     /**
      * Adds the given account to this wallet if it doesn't contain an account with the same
      * address.
      *
      * @param account The account to add.
-     * @return true if the account was added, false if an account with the same address was already
-     * in the wallet.
+     * @return This wallet instance.
      */
-    public boolean addAccount(Account account) {
-        if (this.accounts.containsKey(account.getScriptHash())) {
-            return false;
+    public Wallet addAccount(Account account) {
```

Comment:
You can remove the addAccount(...) method. We normally only provide the plural version of such methods.